File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    let a = [1,2,3,4,5]
    let b = [2,1,2,3,2,4,2,5]
    let c = [3,3,1,1,2,2,4,4,5,5]
    let res = [0,0,0]
    for(let i=0; i<answers.length; i++){
        if(a[i%5]===answers[i]) res[0]++;
        if(b[i%8]===answers[i]) res[1]++;
        if(c[i%10]===answers[i]) res[2]++;
        
    }
    let max = Math.max(...res)
    let answer = []
    for(let i=0; i<res.length; i++){
        if(res[i]===max)
            answer.push(i+1)
    }
    return answer
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    cnt = [0,0,0]
    a1 = [1,2,3,4,5]
    a2 = [2,1,2,3,2,4,2,5]
    a3 = [3,3,1,1,2,2,4,4,5,5]
    for i in range(len(answers)):
        if a1[i%5] == answers[i]:
            cnt[0] += 1
        if a2[i%8] == answers[i]:
            cnt[1] += 1
        if a3[i%10] == answers[i]:
            cnt[2] += 1
    answer = []
    for i in range(len(cnt)):
        if cnt[i] == max(cnt):
            answer.append(i+1)
    return answer